### DISCUSSION TITLE
`@next/next/no-server-import-in-page` should allow `import type {...}`

### DISCUSSION THREAD

**Discussion by mikestopcontinues** [vercel/next.js#33093] [Ideas]
### Describe the feature you'd like to request

Type imports should be allowed outside of `_middleware`, so that you can write reusable helpers (e.g. `(req: NextRequest, res: NextResponse) => NextResponse`) and import them where needed.

### Describe the solution you'd like

Just don't blow this whistle.

### Describe alternatives you've considered

Nothing I can think of...?

**Comment by balazsorban44** [comment]
If it does not guide you, you can just turn that particular rule off. In most cases though, it is probably helpful to people.

**Reply by mikestopcontinues** [reply]
Still, it's flagging an incorrect line of code. `import type {NextRequest} from 'next/server'` doesn't actually do anything problematic, but a non-type import is definitely worth flagging.
